File: 1013-FibonacciNumber/1013-FibonacciNumber.py
# Last updated: 6/25/2026, 9:12:12 AM
class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        memo={}
        return self.fib_memo(n,memo)

    def fib_memo(self,n,memo):
        if n<=1:
            return n
        if n in memo:
            return memo[n]
        memo[n]=self.fib_memo(n-1,memo) + self.fib_memo(n-2,memo)
        return memo[n]

File: 1013-FibonacciNumber/1013-FibonacciNumber.cpp
// Last updated: 6/25/2026, 9:12:17 AM
class Solution {
public:
    int fib(int n) {
        // DP
        if(n==0) return 0;
        if(n==1) return 1;
        int prev1=1, prev2=0, result;
        for(int i=2; i<=n; i++){
            result = prev1+prev2;
            prev2=prev1;
            prev1=result;
        }
        return result;
    }
};